test/utils/testHelper.ts

```diff
@@ -109,6 +135,17 @@ function mockAppConfigurationClientGetConfigurationSetting(kvList) {
     });
 }
 
+function mockAppConfigurationClientListConfigurationSettingsWithFailure(...pages: ConfigurationSetting[][]) {
+    const stub = sinon.stub(AppConfigurationClient.prototype, "listConfigurationSettings");
+
+    // Configure the stub to throw an error on the first call and return mockedKVs on the second call
+    stub.onFirstCall().throws(new RestError("Internal Server Error", { statusCode: 500 }));

```

Comment:
Instead of stubing the listConfigurationSettings method for the `AppConfigurationClient.prototype`,  it is better to stub the client instances you created in mockGetClients.